pkg/crc/machine/start.go

```diff
@@ -828,52 +825,15 @@ func logBundleDate(crcBundleMetadata *bundle.CrcBundleInfo) {
 	}
 }
 
-func ensureRoutesControllerIsRunning(sshRunner *crcssh.Runner, ocConfig oc.Config, preset crcPreset.Preset, bundleInfo *bundle.CrcBundleInfo) error {
+func ensureRoutesControllerIsRunning(sshRunner *crcssh.Runner, ocConfig oc.Config) error {
 	// Check if the bundle have `/opt/crc/routes-controller.yaml` file and if it has
 	// then use it to create the resource for the routes controller.
-	if _, _, err := sshRunner.Run("ls", "/opt/crc/routes-controller.yaml"); err == nil {
-		_, _, err := ocConfig.RunOcCommand("apply", "-f", "/opt/crc/routes-controller.yaml")
-		return err
-	}
-	// TODO: Remove this resource creation after crc 2.24 release
-	bin, err := json.Marshal(v1.Pod{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Pod",
-			APIVersion: "v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "routes-controller",
-			Namespace: "openshift-ingress",
-		},
-		Spec: v1.PodSpec{
-			ServiceAccountName: "router",
-			Containers: []v1.Container{
-				{
-					Name:            "routes-controller",
-					Image:           getRouterControllerImage(preset, bundleInfo),
-					ImagePullPolicy: v1.PullIfNotPresent,
-				},
-			},
-		},
-	})
-	if err != nil {
-		return err
-	}
-	if err := sshRunner.CopyData(bin, "/tmp/routes-controller.json", 0644); err != nil {
-		return err
-	}
-	_, _, err = ocConfig.RunOcCommand("apply", "-f", "/tmp/routes-controller.json")
+	_, _, err := sshRunner.Run("ls", "/opt/crc/routes-controller.yaml")
```

Comment:
This could also be dropped, I guess it's checking for the existence of `routes-controller.yaml`. I expect `oc apply` will exit with an error if the file does not exist anyway.

Reply:
yes this can be removed as well, i thought it'd be useful have this check before invoking `oc` but since `oc` would return a similar error its not needed to have this check